src/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC, ReactElement } from 'react';
+import { StyledFormLabel, StyledInput, StyledTextInput } from './TextInput.styles';
+
+type HTMLInputProps = Omit<React.HTMLProps<HTMLInputElement>, 'ref' | 'as'>;
```

Comment:
Could not find the right types for the input element returned by styled-components. Any way to improve this?

Reply:
Can you define the type as the following:

```typescript
export type TextInputProps = {
  label: string;
  type?: string;
  hasError?: boolean;
  hideLabel?: boolean;
} & typeof StyledTextInput;
```

Reply:
I tried this but it won't give me the right input element props. My solution works but I find it strange I have to omit certain values.